### PR TITLE
Development Process Documentation

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in documentation for our development process and additionally adds in a pull request template. The purpose of this is to clarify the process for all team members, and to set out written expectations. These documents were heavily inspired by GitLab.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

Since these are changes to the documentation you can review the updated site with the following steps:
1. `cd docs-site && pnpm run start`
2. Navigate to `http://localhost:4000/irida-next/docs/development/development_processes`
3. Review the two pages linked under that section.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the PR acceptance checklist for this PR.

### CODE REVIEW

File: docs-site/docs/development/development_processes/code_review.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+This guide contains advice and best practices for performing code review, and having your code reviewed.
+
+All merge requests for IRIDA Next, whether written by a IRIDA team member or a wider community member, must go through a code review process to ensure the code is effective, understandable, maintainable, and secure.
+
+## Getting your pull request reviewed, approved, and merged
+
+Before you begin:
+* Familiarize yourself with the [contribution acceptance criteria](./pull_request_workflow/)#contribution-acceptance-criteria
```
Comment (joshsadam) [inline docs-site/docs/development/development_processes/code_review.md:14]: Closing bracket in wrong position

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+### Testing
+
+1. Unit, integration, and system tests that all pass on the CI server.
+1. Regressions and bugs are covered with tests that reduce the risk of the issue happening again.
+1. If your merge request adds one or more migrations, write tests for more complex migrations.
+
+### UI changes
+
+1. Use available components from the IRIDA Next Design System, Viral.
```
Comment (joshsadam) [inline docs-site/docs/development/development_processes/pull_request_workflow.md:66]: The only thing I can think of is if a component does not exist, create a new PR with that component first.

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+---
+
+We welcome pull requests from everyone, with fixes and improvements to IRIDA Next code, tests, and documentation.
+
+## Merge request guidelines for contributors
+
+### Best practices
+
+* If the change is non-trivial, we encourage you start a discussion with a team member. You can do this by tagging them in a PR before submitting the code for review. Taking to team embers can be helpful when making design decisions. Communicating the intent behind your changes can also help expedite pull request reviews.
+* When having your code reviewed and when reviewing pull requests, please keep the [code review guidelines](./code_review) in mind.
```
Comment (joshsadam) [inline docs-site/docs/development/development_processes/pull_request_workflow.md:14]: This link appears to be broken for me.

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks really good, and should really help our development team.  I found a couple links to be broken (noted as comments) and one suggestion for the UI section.  I do think that we should add something for commit messages like https://www.conventionalcommits.org/en/v1.0.0/#summary

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+1. If the PR changes CSS classes, please include the list of affects pages, which can be found by running `grep css-class ./app -R`.
+
+### Description of changes
+
+1. Clear title and description explaining the relevancy of the contribution.
+1. Description includes any steps or setup required to ensure reviewers can view the changes you've made.
+
+### Approval
+
+1. The [PR acceptance checklist](./code_review#acceptance-checklist) has been checked as confirmed in the PR.
```
[the remark this answers is not in the document]
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md:77]: Fixed in 23599ec0. I forgot to save and push the file...

File: docs-site/docs/development/development_processes/pull_request_workflow.md (hunk 3 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md:14]: Can you elaborate, are you saying when you run the docs site that the link doesn't work? The link works normally from me when running the docs site with `pnpm run start` inside of the `docs-site` folder.

File: docs-site/docs/development/development_processes/code_review.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+### Acceptance checklist
+
+This checklist encourages the authors, reviewers, and maintainers of pull requests (MRs) to confirm changes were analyzed for high-impact risks to quality, performance, reliability, security, observability, and maintainability.
+
+Using checklists improves quality in software engineering. This checklist is a straightforward tool to support and bolster the skills of contributors to the IRIDA Next codebase.
+
+#### Quality
+
+1. You have self-reviewed this PR per [code review guidelines](./code_review).
+1. For the code that this change impacts, you beloeve that the automated tests validated functionality that is highly important to users.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/code_review.md:31]: Typo in `believe` and add a comma after `you believe that`

File: docs-site/docs/development/development_processes/code_review.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+1. You have added information for database reviewers in the PR description, or you have decided that it is unnecessary.
+1. You have considered the scalability risk based on future predicted growth.
+
+#### Documentation
+
+1. You have added/updated documentation or decided that documentation changes are unnecessary for this PR.
+
+#### Security
+
+1. You have confirmed that if thsi PR contains a change to processing or storing credentials or tokens, authorization and authentication methods, you have added the `~security` label.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/code_review.md:48]: Typo in `this`

File: docs-site/docs/development/development_processes/code_review.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+* Accept that many programming decisions are opinions. Discuss tradeoffs, which you prefer, and reach a resolution quickly.
+* Ask questions; don't make demands. ("What do you think about naming this `:sample_id`?")
+* Ask for clarification. ("I didn't understand. Can you clarify?")
+* Avoid selective ownership of code. ("mine", "not mine", "yours")
+* Avoid using terms that could be seen as referring to personal traits. ("dumb", "stupid"). Assume everyone is intelligent and well-meaning.
+* Be explicit. Remember people don't always understand your intentions online.
+* Be humble. ("I'm not sure - let's look it up.")
+* Don't use hyperbole. ("always", "never", "endlessly", "nothing")
+* Be careful about the use of sarcasm. Everything we do is public; what seems like good-natured ribbing to you and a long-time colleague might come off as mean and unwelcoming to a person new to the project.
+* Consider one-on-one chats or video valls if there are too many "I didn't understand" or "Alternative solution:" comments. Post a follow-up comment summarizing one-on-one discussion.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/code_review.md:64]: Typo in `calls`

File: docs-site/docs/development/development_processes/code_review.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+* If you ask a question to a specific person, always start the comment by mentioning them.
+
+### Having your pull request reviewed
+
+Please keep in mind that code review is a process that can take multiple iterations, and reviewers may spot things later that they may not have seen the first time.
+
+* The first reviewer of your code is *you*. Before you perform that first push of your shiny new branch, read through the entire diff. Does it make sense? Did you include something unrelated to the overall purpose of the changes? Did you forget to remove any debugging code?
+* Write a detailed description as outlined in the [pull request guidelines](./pull_request_workflow). SOme reviewers may not be familiar with the feature or area of the codebase. Thorough descriptions help all reviewers understand your request and test effectively.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/code_review.md:72]: Double capitals in `Some`

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+1. Unit, integration, and system tests that all pass on the CI server.
+1. Regressions and bugs are covered with tests that reduce the risk of the issue happening again.
+1. If your merge request adds one or more migrations, write tests for more complex migrations.
+
+### UI changes
+
+1. Use available components from the IRIDA Next Design System, Viral.
+1. The PR must include *Before* and *After* screenshots if UI changes are made.
+1. If the PR changes CSS classes, please include the list of affects pages, which can be found by running `grep css-class ./app -R`.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/pull_request_workflow.md:68]: Changed `affects` to `affected`

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+### Description of changes
+
+1. Clear title and description explaining the relevancy of the contribution.
+1. Description includes any steps or setup required to ensure reviewers can view the changes you've made.
+
+### Approval
+
+1. The [PR acceptance checklist](./code_review#acceptance-checklist) has been checked as confirmed in the PR.
+1. Reviewed by relevant reviewers, and all concerns are addressed for Availability, Regressions, and Security. Documentation reviews should take place as soon as possible, but they should not block a merge request.
+1. You pull request has at least 1 approval, but depending on your changes might need additional approvals.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/pull_request_workflow.md:79]: Change `You` to `Your`

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  * You don't have to select any specific reviewers, but you can if you really want specific people to approve your pull request.
+1. Merged by a project maintainer.
```
Comment (deepsidhu85) [inline docs-site/docs/development/development_processes/pull_request_workflow.md:81]: All the other items are numbered correctly when viewing on the docs site but for some reason this last one shows up with a `1.`

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks great just a few small things below

File: docs-site/docs/development/development_processes/pull_request_workflow.md (hunk 11 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md:81]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

File: docs-site/docs/development/development_processes/pull_request_workflow.md (hunk 10 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

File: docs-site/docs/development/development_processes/pull_request_workflow.md (hunk 9 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

File: docs-site/docs/development/development_processes/code_review.md (hunk 7 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/code_review.md]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

File: docs-site/docs/development/development_processes/code_review.md (hunk 5 above)
Comment (ericenns) [reply docs-site/docs/development/development_processes/code_review.md]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

Leaving out comma change

File: docs-site/docs/development/development_processes/pull_request_workflow.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+## Contribution acceptance criteria
+
+To make sure that your pull request can be approved, please ensure that it meets the contribution acceptance criteria below:
+
+1. The change is as small as possible.
+1. If the pull request contains more than 500 changes:
+   * Explain the reason
+1. Mention any major breaking changes.
+1. Inlcude proper test and make sure all tests pass (unless it contains a test exposing a bug in existing code). Every new class should have corresponding unit tests, even if the class is exercised at a higher level, such as a feature test.
```
[the remark this answers is not in the document]
Comment (ericenns) [reply docs-site/docs/development/development_processes/pull_request_workflow.md]: Addressed in https://github.com/phac-nml/irida-next/pull/105/commits/aa95bea4d612c311ca86792f706dd567be19251c

**Review by deepsidhu85 — APPROVED** [review]

**Comment by ericenns** [comment]
> This looks really good, and should really help our development team. I found a couple links to be broken (noted as comments) and one suggestion for the UI section. I do think that we should add something for commit messages like https://www.conventionalcommits.org/en/v1.0.0/#summary

In discussion with @deepsidhu85, we decided that addition of commit guidelines should be completed in a follow up PR.

**Review by joshsadam — APPROVED** [review]
I cannot say again how much I love what is covered in this PR.  Thanks Eric

**Pull request merged**